forte/utils/tagging_scheme.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2021 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utility functions related to tagging scheme.
+"""
+
+import warnings
+from typing import Optional, List, Union, Tuple, Any
+
+
+def bio_merge(tags: List[str], types: List[Union[str, None]],
+              start: Optional[List[int]]=None, end: Optional[List[int]]=None) \
+        -> Tuple[List[Any], List[Any], List[Any]]:
+    r""" This function merged BIO-schemed augmented tagging scheme results and
+    return entity mentions information.
+
+    For example, BIO NER tags could be merged by passing
+    tags = ['B', 'O', 'B', 'I']
+    types = ['PER', '', 'LOC', 'LOC']
+    start = [0, 11, 20, 24]
+    end = [1, 19, 22, 26]
+
+    After merging BIO tags, the result will be returned as
+    result_types = ['PER', 'LOC']
+    result_start = [0, 20]
+    result_end = [1, 26]
+
+    The function can also handle tags with no types, for example, in some word
+    segmentation tasks. In this case the input `types` should be set as a list
+    of None, and the returned result_type will be a list of None.
+
+
```

Comment:
Looks like this function handles incorrect tag sequences, can you document how will they be handled?